hs_core/views/utils.py

```diff
@@ -659,6 +660,25 @@ def remove_folder(user, res_id, folder_path):
     hydroshare.utils.resource_modified(resource, user)
 
 
+def list_folder(res_id, folder_path):
+    """
+    list a sub-folder/sub-collection in hydroshareZone or any federated zone used for HydroShare
+    resource backend store.
+    :param user: requesting user
+    :param res_id: resource uuid
+    :param folder_path: the relative path for the folder to be listed under res_id collection.
+    :return:
+    """
```

Comment:
@alvacouch This added ```list_folder``` function is an extraction of the code in this back end view function which is used to serve AJAX calls made from UI folder browsing: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/views/resource_folder_hierarchy.py#L19..L82

Now that you added this utility function, you can refactor that data_store_structure() view function to call this function instead to reduce code duplication.